internal/pkg/service/buffer/store/model/model.go

```diff
@@ -62,6 +62,40 @@ type ImportConditions struct {
 	Time  time.Duration     `json:"time" validate:"min=30000000000,max=86400000000000"` // 30s-24h
 }
 
+func (c ImportConditions) ShouldImport(s CurrentImportState) (bool, string) {
+	if s.Count == 0 {
+		return false, "no data to import"
+	}
+
+	defaults := DefaultConditions()
+	if c.Count == 0 {
+		c.Count = defaults.Count
+	}
+	if c.Size == 0 {
+		c.Size = defaults.Size
+	}
+	if c.Time == 0 {
+		c.Time = defaults.Time
+	}
+
+	if s.Count >= c.Count {
+		return true, fmt.Sprintf("import count met, imported: %d rows, limit: %d rows", s.Count, c.Count)
+	}
+	if s.Size >= c.Size {
+		return true, fmt.Sprintf("import size met, imported: %s, limit: %s", s.Size.String(), c.Size.String())
+	}
+	if s.Time >= c.Time {
+		return true, fmt.Sprintf("import time met, last import at: %s, limit: %s", s.Time.String(), c.Time.String())
+	}
```

Comment:
@michaljurecko Nějak mi ten `ConditionsResult` co jsi navrhoval asi nakonec moc nedává smysl. Resp. bych vytvořil strukturu s fieldama `bool` a `string` což můžu stejně dobře vrátit takto, protože ty metody by na ní asi neměly co dělat? Nebo jsi to myslel tak, že bych tady jen vytvořil tu strukturu a resolvování conditions přenesl do ní?

```go
func (c ImportConditions) ShouldImport(s CurrentImportState) (bool, string) {
    return ConditionsResult{State: s, Conditions: c}
}
```

Reply:
nemam presnu predstavu, ... ciel je dostat z toho string, ktory sa da zalogova `starting upload <REASON>`, a poznamka bokom bola, ze mozno to dava zmysel ako struktura so `String()` method.